// notistack/build/SnackbarContainer/SnackbarContainer.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _styles = require('@material-ui/core/styles');

var _SnackbarContainer = require('./SnackbarContainer.styles');

var _SnackbarContainer2 = _interopRequireDefault(_SnackbarContainer);

var _constants = require('../utils/constants');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var SnackbarContainer = function SnackbarContainer(_ref) {
    var _classNames;

    var children = _ref.children,
        classes = _ref.classes,
        renderVariant = _ref.renderVariant,
        anchorOrigin = _ref.anchorOrigin;

    var className = (0, _classnames2.default)((_classNames = {}, _defineProperty(_classNames, classes.root, true), _defineProperty(_classNames, classes.reverseColumns, anchorOrigin.vertical === 'bottom'), _defineProperty(_classNames, classes.top, anchorOrigin.vertical === 'top'), _defineProperty(_classNames, classes.bottom, anchorOrigin.vertical === 'bottom'), _defineProperty(_classNames, classes.left, anchorOrigin.horizontal === 'left'), _defineProperty(_classNames, classes.center, anchorOrigin.horizontal === 'center'), _defineProperty(_classNames, classes.right, anchorOrigin.horizontal === 'right'), _classNames));
    return renderVariant === _constants.RENDER_VARIANTS.wrapped ? _react2.default.createElement(
        'div',
        { className: className },
        children
    ) : children;
};

SnackbarContainer.propTypes = {
    children: _propTypes2.default.node,
    classes: _propTypes2.default.object,
    renderVariant: _propTypes2.default.oneOf(Object.keys(_constants.RENDER_VARIANTS)),
    anchorOrigin: _propTypes2.default.shape({
        horizontal: _propTypes2.default.oneOf(['left', 'center', 'right']).isRequired,
        vertical: _propTypes2.default.oneOf(['top', 'bottom']).isRequired
    })
};

exports.default = (0, _styles.withStyles)(_SnackbarContainer2.default)(SnackbarContainer);
